src/Compilers/CSharp/Test/Syntax/Parsing/CSharpParseOptionsTests.cs

```diff
@@ -22,6 +22,17 @@ private void TestProperty<T>(Func<CSharpParseOptions, T, CSharpParseOptions> fac
         }
 
         [Fact]
+        [WorkItem(15358, "https://github.com/dotnet/roslyn/issues/15358")]
+        public void WithDocumentationModeDoesntChangeFeatures()
+        {
+            var kvp = new KeyValuePair<string, string>("IOperation", "true");
+            var po = new CSharpParseOptions().WithFeatures(new[] { kvp });
+            Assert.Equal(po.Features.AsSingleton(), kvp);
+            var po2 = po.WithDocumentationMode(DocumentationMode.Diagnose);
```

Comment:
>var po2 = po.WithDocumentationMode(DocumentationMode.Diagnose); [](start = 12, length = 63)

I think it would be good to assert DocumentationMode before and after. What if the WithDocumentationMode method does nothing?